test/src/modules/navigation_tests.coffee

```diff
@@ -39,6 +39,14 @@ navigationTest "following a same-origin data-turbolinks=true link inside a data-
     assert.equal(navigation.action, "push")
     done()
 
+navigationTest "following a same-origin link with ID hash", (assert, session, done) ->
```

Comment:
Turbolinks uses the term "anchor" internally. Let's stick with that convention and call this "following a same-origin anchored link" and `same-origin-anchored-link`.